block.go

```diff
@@ -0,0 +1,53 @@
+package sdk
+
+import (
+	"context"
+
+	ctypes "github.com/cometbft/cometbft/rpc/core/types"
+	cosmos "github.com/cosmos/cosmos-sdk/client"
+)
+
+// NodeStatusFetcher returns the status of the node to which it is connected.
+//
+// This status is used here in the BlockClient in order to get the latest block height.
+type NodeStatusFetcher interface {
+	Status(ctx context.Context) (*ctypes.ResultStatus, error)
+}
+
+// BlockClient is the interface to interact with the on-chain block module.
+//
+// For example, it can be used to get the latest block height.
+//
+// It uses the StatusClient interface of the cometbft to pull the latest block height.
+// This is done to avoid fetching the entire latest block just to extract the block height.
+//
+// The implementations of this interface could leverage websockets to get notified
+// about new blocks and cache the latest block height to avoid querying the blockchain
+// for it every time.
+type BlockClient struct {
+	NodeStatusFetcher
```

Comment:
I'm still confused by why we have a publically exposed interface AND a publically exposed struct. 

As a consumer, I'm not sure which I should use.

Reply:
Thank you, addressed:
- New file `block_test.go` has an example usage.
- Description on the interface copy/pasted here:
```
// PoktNodeStatusFetcher interface is used by the BlockClient to get the status of a POKT full node.
// The BlokClient extracts the latest height from this status struct.
//
// Most users can rely on the default implementation provided by NewPoktNodeStatusFetcher function.
// A custom implementation of this interface can be used to gain more granular control over the interactions
// of the BlockClient with the POKT full node.
```